_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,374 @@
+For example, in a cluster of 5 brokers the pods are named `<CLUSTER-NAME>-kafka-0` up to `<CLUSTER-NAME>-kafka-4`.
+If you decide to scale down by two brokers, then `<CLUSTER-NAME>-kafka-4` and `<CLUSTER-NAME>-kafka-3` will be removed.
+
+ The next section will help you set up the environment for executing the above example, i.e. setting up your kafka cluster, kafka topics and also configuring the Kafka user with required ACL's.
+
+Note: In case you already have the environment set up, you can skip the next section.
+
+### Setting up the environment
+
+To get the Kafka cluster up and running, we will first have to install the Strimzi Cluster Operator and then deploy the Kafka resource. You can refer to the [Stimzi Quickstart Guide](https://strimzi.io/docs/operators/latest/quickstart.html) for installing Strimzi.
+
+You can install the Cluster Operator with any installation method you prefer.
+The Kafka cluster is then deployed with the plain listener enabled on port 9092.
+
+Example Kafka configuration with plain listener.
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+   name: my-cluster
+spec:
+   kafka:
+      version: 3.2.0
+      replicas: 5
+      listeners:
+         - name: plain
+           port: 9092
+           type: internal
+           tls: false
+         - name: tls
+           port: 9093
+           type: internal
+           tls: true
+      config:
+         offsets.topic.replication.factor: 3
+         transaction.state.log.replication.factor: 3
+         transaction.state.log.min.isr: 2
+         default.replication.factor: 3
+         min.insync.replicas: 2
+         inter.broker.protocol.version: "3.2"
+      storage:
+         type: ephemeral
+   zookeeper:
+      replicas: 3
+      storage:
+         type: ephemeral
+   entityOperator:
+      topicOperator: {}
+      userOperator: {}
+```
+
+Once the cluster is running, let's deploy some topics where we will send and receive the messages.
+Here is an example topic configuration. You can change it to suit your requirements.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic-two
+  labels:
+    strimzi.io/cluster: <CLUSTER-NAME>
+spec:
+  partitions: 10
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+    # ...
+```
+
+### Creating a proposal `reassignment.json` file
+
+When you have a Kafka cluster running with brokers and topics, you are ready to create the proposal JSON file.
+
+Let us create a separate interactive pod.
+This interactive pod is used to run all the reassignment commands. 
+One question might bug you. What is the need of a separate pod? Can't we just use one of the broker pods?
+In answer to this question, running commands from within a broker is not good practice.
+It will start another JVM inside the container designated for the broker and can cause disruption, cause the container to run out of memory and so on.
```

Comment:
```suggestion
Running any of the Kafka `/bin` scripts from within the broker container will start another JVM (with all the same settings as the Kafka broker).
This can cause disruption, including causing the container to run out of memory.
```